File: .sling/libs/web5/eqela/jk-data/0.32.0/jk.model.js
let JkModelDynamicModel = function() {
	this.dataValue = null;
	this.fieldsValue = null;
};

JkModelDynamicModel.prototype._t = {
	"JkJsonJSONObject" : true,
	"JkModelDynamicModel" : true,
	"JkLangStringObject" : true
};
JkModelDynamicModel.prototype._tobj = JkModelDynamicModel;

JkModelDynamicModel.NEW_JkModelDynamicModelJkLangDynamicMap = function(data, mapping) {
	var v = new JkModelDynamicModel;
	return v.CTOR_JkModelDynamicModel_JkModelDynamicModelJkLangDynamicMap(data, mapping);
};

JkModelDynamicModel.prototype.CTOR_JkModelDynamicModel_JkModelDynamicModelJkLangDynamicMap = function(data, mapping) {
	this.fieldsValue = null;
	this.dataValue = null;
	this.setData(null);
	this.copyFrom(data, mapping);
	return this;
};

JkModelDynamicModel.NEW_JkLangDynamicMap = function(data) {
	var v = new JkModelDynamicModel;
	return v.CTOR_JkModelDynamicModel_JkLangDynamicMap(data);
};

JkModelDynamicModel.prototype.CTOR_JkModelDynamicModel_JkLangDynamicMap = function(data) {
	this.fieldsValue = null;
	this.dataValue = null;
	this.setData(data);
	return this;
};

JkModelDynamicModel.NEW = function() {
	var v = new JkModelDynamicModel;
	return v.CTOR_JkModelDynamicModel();
};

JkModelDynamicModel.prototype.CTOR_JkModelDynamicModel = function() {
	this.fieldsValue = null;
	this.dataValue = null;
	this.setData(null);
	return this;
};

JkModelDynamicModel.asDynamicMap = function(object) {
	if(!(object != null)) {
		return null;
	}
	return object.toDynamicMap();
};

JkModelDynamicModel.prototype.toString = function() {
	return JkJsonJSONEncoder.toString((this.toDynamicMap()));
};

JkModelDynamicModel.prototype.toDynamicMap = function() {
	return this.getData();
};

JkModelDynamicModel.prototype.fromDynamicMap = function(data) {
	this.setData(data);
};

JkModelDynamicModel.prototype.toJsonObject = function() {
	return this.toDynamicMap();
};

JkModelDynamicModel.prototype.setField = function(name, value) {
	this.getData().setObject(name, value);
};

JkModelDynamicModel.prototype.getField = function(name) {
	return this.getData().get(name);
};

JkModelDynamicModel.prototype.getOrCreateFieldAsDynamicVector = function(name) {
	var v = this.getData().getDynamicVector(name);
	if(!(v != null)) {
		v = JkLangDynamicVector.NEW();
		this.getData().setObject(name, v);
	}
	return v;
};

JkModelDynamicModel.prototype.getFieldAsDynamicVector = function(name) {
	return this.getData().getDynamicVector(name);
};

JkModelDynamicModel.prototype.getFieldAsStringVector = function(name) {
	var v = this.getFieldAsDynamicVector(name);
	if(!(v != null)) {
		return null;
	}
	return v.toVectorOfStrings();
};

JkModelDynamicModel.prototype.getFieldAsIntegerVector = function(name) {
	var vector = this.getFieldAsDynamicVector(name);
	if(!(vector != null)) {
		return null;
	}
	var v = new Array;
	{
		var _vl = vector;
		if(_vl != null) {
			var _it = _vl.iterate();
			while(_it != null && _it.hasNext()){
				var o = _it.next();
				if(o == null) {
					continue;
				}
				v.push(JkLangInteger.asInteger(o));
			}
		}
	}
	return v;
};

JkModelDynamicModel.prototype.getFieldAsLongIntegerVector = function(name) {
	var vector = this.getFieldAsDynamicVector(name);
	if(!(vector != null)) {
		return null;
	}
	var v = new Array;
	{
		var _vl = vector;
		if(_vl != null) {
			var _it = _vl.iterate();
			while(_it != null && _it.hasNext()){
				var o = _it.next();
				if(o == null) {
					continue;
				}
				v.push(JkLangLongInteger.asLong(o));
			}
		}
	}
	return v;
};

JkModelDynamicModel.prototype.getFieldAsDoubleVector = function(name) {
	var vector = this.getFieldAsDynamicVector(name);
	if(!(vector != null)) {
		return null;
	}
	var v = new Array;
	{
		var _vl = vector;
		if(_vl != null) {
			var _it = _vl.iterate();
			while(_it != null && _it.hasNext()){
				var o = _it.next();
				if(o == null) {
					continue;
				}
				v.push(JkLangDouble.asDouble(o));
			}
		}
	}
	return v;
};

JkModelDynamicModel.prototype.getFieldAsBooleanVector = function(name) {
	var vector = this.getFieldAsDynamicVector(name);
	if(!(vector != null)) {
		return null;
	}
	var v = new Array;
	{
		var _vl = vector;
		if(_vl != null) {
			var _it = _vl.iterate();
			while(_it != null && _it.hasNext()){
				var o = _it.next();
				if(o == null) {
					continue;
				}
				v.push(JkLangBoolean.asBoolean(o, false));
			}
		}
	}
	return v;
};

JkModelDynamicModel.prototype.addStringToVectorField = function(field, value) {
	this.getOrCreateFieldAsDynamicVector(field).appendString(value);
};

JkModelDynamicModel.prototype.addIntegerToVectorField = function(field, value) {
	this.getOrCreateFieldAsDynamicVector(field).appendInteger(value);
};

JkModelDynamicModel.prototype.addLongIntegerToVectorField = function(field, value) {
	this.getOrCreateFieldAsDynamicVector(field).appendLong(value);
};

JkModelDynamicModel.prototype.addDoubleToVectorField = function(field, value) {
	this.getOrCreateFieldAsDynamicVector(field).appendDouble(value);
};

JkModelDynamicModel.prototype.addBooleanToVectorField = function(field, value) {
	this.getOrCreateFieldAsDynamicVector(field).appendBoolean(value);
};

JkModelDynamicModel.prototype.addBufferToVectorField = function(field, value) {
	this.getOrCreateFieldAsDynamicVector(field).appendObject(value);
};

JkModelDynamicModel.prototype.addObjectToVectorField = function(field, value) {
	var vv = this.getOrCreateFieldAsDynamicVector(field);
	if(value != null && (JkModelDynamicModel.IS_INSTANCE && JkModelDynamicModel.IS_INSTANCE(value))) {
		vv.appendObject((value.toDynamicMap()));
	}
	else {
		vv.appendObject(value);
	}
};

JkModelDynamicModel.prototype.fromJsonObject = function(o1) {
	var data = (function(o) {
		if(JkLangDynamicMap.IS_INSTANCE && JkLangDynamicMap.IS_INSTANCE(o)) {
			return o;
		}
		return null;
	}.bind(this))(o1);
	if(data != null) {
		this.fromDynamicMap(data);
	}
	else {
		this.clear();
	}
	return true;
};

JkModelDynamicModel.prototype.containsField = function(field) {
	if(!(field != null)) {
		return false;
	}
	var array = this.getFields();
	if(array != null) {
		var n = 0;
		var m = array.length;
		for(n = 0 ; n < m ; n++) {
			var ff = array[n];
			if(ff != null) {
				if(JkLangString.equals((ff.getName()), field)) {
					return true;
				}
			}
		}
	}
	return false;
};

JkModelDynamicModel.prototype.hasValue = function(key) {
	if(!(key != null)) {
		return false;
	}
	return this.getData().containsKey(key);
};

JkModelDynamicModel.prototype.removeNullValues = function() {
	if(this.dataValue != null) {
		this.dataValue.removeNullValues();
	}
};

JkModelDynamicModel.prototype.clear = function() {
	this.dataValue = null;
};

JkModelDynamicModel.prototype.setAllFields = function() {
	var array = this.getFields();
	if(array != null) {
		var n = 0;
		var m = array.length;
		for(n = 0 ; n < m ; n++) {
			var field = array[n];
			if(field != null) {
				if(!this.getData().containsKey((field.getName()))) {
					this.getData().setObject((field.getName()), null);
				}
			}
		}
	}
};

JkModelDynamicModel.prototype.getModelMetadata = function() {
	return null;
};

JkModelDynamicModel.prototype.getFieldInformation = function() {
	return null;
};

JkModelDynamicModel.prototype.copyFrom = function(other, mapping) {
	if(!(other != null)) {
		return;
	}
	var array = this.getFields();
	if(array != null) {
		var n = 0;
		var m = array.length;
		for(n = 0 ; n < m ; n++) {
			var field = array[n];
			if(field != null) {
				var fname = field.getName();
				var oname = null;
				if(mapping != null && mapping.containsKey(fname)) {
					oname = mapping.getString(fname, null);
				}
				else {
					oname = fname;
				}
				if(oname != null && other.hasValue(oname)) {
					this.setField(fname, (other.getField(oname)));
				}
			}
		}
	}
};

JkModelDynamicModel.prototype.getData = function() {
	if(!(this.dataValue != null)) {
		this.dataValue = JkLangDynamicMap.NEW();
	}
	return this.dataValue;
};

JkModelDynamicModel.prototype.doSetData = function(value) {
	this.dataValue = value;
};

JkModelDynamicModel.prototype.setData = function(v) {
	this.doSetData(v);
	return v;
};

JkModelDynamicModel.prototype.getFields = function() {
	if(!(this.fieldsValue != null)) {
		this.fieldsValue = new Array;
		var array = this.getFieldInformation();
		if(array != null) {
			var n = 0;
			var m = array.length;
			for(n = 0 ; n < m ; n++) {
				var name = array[n];
				if(name != null) {
					this.fieldsValue.push(name);
				}
			}
		}
	}
	return this.fieldsValue;
};

JkModelDynamicModel.prototype.doSetFields = function(value) {
	this.fieldsValue = value;
};

JkModelDynamicModel.prototype.setFields = function(v) {
	this.doSetFields(v);
	return v;
};

JkModelDynamicModel.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkModelDynamicModel"] === true;
};

let JkModelDynamicModelField = function() {
	this._name = null;
	this._encodedName = null;
	this._type = 0;
	this._description = null;
	this._tags = null;
};

JkModelDynamicModelField.prototype._t = { "JkModelDynamicModelField" : true };
JkModelDynamicModelField.prototype._tobj = JkModelDynamicModelField;

JkModelDynamicModelField.NEW = function() {
	var v = new JkModelDynamicModelField;
	return v.CTOR_JkModelDynamicModelField();
};

JkModelDynamicModelField.prototype.CTOR_JkModelDynamicModelField = function() {
	this._tags = null;
	this._description = null;
	this._type = 0;
	this._encodedName = null;
	this._name = null;
	return this;
};

JkModelDynamicModelField.prototype.hasTag = function(tag) {
	var array = this.getTags();
	if(array != null) {
		var n = 0;
		var m = array.length;
		for(n = 0 ; n < m ; n++) {
			var tt = array[n];
			if(tt != null) {
				if(tt == tag) {
					return true;
				}
			}
		}
	}
	return false;
};

JkModelDynamicModelField.prototype.addToTags = function(tag) {
	if(!(tag != null)) {
		return;
	}
	if(!(this.getTags() != null)) {
		this.setTags((new Array));
	}
	this.getTags().push(tag);
};

JkModelDynamicModelField.prototype.getName = function() {
	return this._name;
};

JkModelDynamicModelField.prototype.doSetName = function(value) {
	this._name = value;
};

JkModelDynamicModelField.prototype.setName = function(v) {
	this.doSetName(v);
	return v;
};

JkModelDynamicModelField.prototype.getEncodedName = function() {
	return this._encodedName;
};

JkModelDynamicModelField.prototype.doSetEncodedName = function(value) {
	this._encodedName = value;
};

JkModelDynamicModelField.prototype.setEncodedName = function(v) {
	this.doSetEncodedName(v);
	return v;
};

JkModelDynamicModelField.prototype.getType = function() {
	return this._type;
};

JkModelDynamicModelField.prototype.doSetType = function(value) {
	this._type = value;
};

JkModelDynamicModelField.prototype.setType = function(v) {
	this.doSetType(v);
	return v;
};

JkModelDynamicModelField.prototype.getDescription = function() {
	return this._description;
};

JkModelDynamicModelField.prototype.doSetDescription = function(value) {
	this._description = value;
};

JkModelDynamicModelField.prototype.setDescription = function(v) {
	this.doSetDescription(v);
	return v;
};

JkModelDynamicModelField.prototype.getTags = function() {
	return this._tags;
};

JkModelDynamicModelField.prototype.doSetTags = function(value) {
	this._tags = value;
};

JkModelDynamicModelField.prototype.setTags = function(v) {
	this.doSetTags(v);
	return v;
};

JkModelDynamicModelField.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkModelDynamicModelField"] === true;
};

JkModelDynamicModelField.TYPE_UNKNOWN = 0;
JkModelDynamicModelField.TYPE_OBJECT = 1;
JkModelDynamicModelField.TYPE_STRING = 2;
JkModelDynamicModelField.TYPE_BUFFER = 3;
JkModelDynamicModelField.TYPE_INTEGER = 4;
JkModelDynamicModelField.TYPE_LONG_INTEGER = 5;
JkModelDynamicModelField.TYPE_DOUBLE = 6;
JkModelDynamicModelField.TYPE_BOOLEAN = 7;
JkModelDynamicModelField.TYPE_VECTOR = 8;
JkModelDynamicModelField.TYPE_MAP = 9;

File: .sling/libs/sushi/eqela/jk-data/0.23.0/jk.model.lua
jk = jk or {}

jk.model = jk.model or {}

jk.model.DynamicModel = {}
jk.model.DynamicModel.__index = jk.model.DynamicModel
_vm:set_metatable(jk.model.DynamicModel, {})

function jk.model.DynamicModel._create()
	local v = _vm:set_metatable({}, jk.model.DynamicModel)
	return v
end

function jk.model.DynamicModel:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.model.DynamicModel'
	self['_isType.jk.model.DynamicModel'] = true
	self['_isType.jk.json.JSONObject'] = true
	self['_isType.jk.lang.StringObject'] = true
	self.dataValue = nil
	self.fieldsValue = nil
end

function jk.model.DynamicModel:asDynamicMap(object)
	if not (object ~= nil) then
		do return nil end
	end
	do return object:toDynamicMap() end
end

function jk.model.DynamicModel:_construct0()
	jk.model.DynamicModel._init(self)
	do self:setData(nil) end
	return self
end

function jk.model.DynamicModel:_construct1(data)
	jk.model.DynamicModel._init(self)
	do self:setData(data) end
	return self
end

function jk.model.DynamicModel:_construct2(data, mapping)
	jk.model.DynamicModel._init(self)
	do self:setData(nil) end
	do self:copyFrom(data, mapping) end
	return self
end

function jk.model.DynamicModel:toString()
	do return _g.jk.json.JSONEncoder:toString(self:toDynamicMap()) end
end

function jk.model.DynamicModel:toDynamicMap()
	do return self:getData() end
end

function jk.model.DynamicModel:fromDynamicMap(data)
	do self:setData(data) end
end

function jk.model.DynamicModel:toJsonObject()
	do return self:toDynamicMap() end
end

function jk.model.DynamicModel:setField(name, value)
	do self:getData():setObject(name, value) end
end

function jk.model.DynamicModel:getField(name)
	do return self:getData():get(name) end
end

function jk.model.DynamicModel:getOrCreateFieldAsDynamicVector(name)
	local v = self:getData():getDynamicVector(name)
	if not (v ~= nil) then
		v = _g.jk.lang.DynamicVector._construct0(_g.jk.lang.DynamicVector._create())
		do self:getData():setObject(name, v) end
	end
	do return v end
end

function jk.model.DynamicModel:getFieldAsDynamicVector(name)
	do return self:getData():getDynamicVector(name) end
end

function jk.model.DynamicModel:getFieldAsStringVector(name)
	local v = self:getFieldAsDynamicVector(name)
	if not (v ~= nil) then
		do return nil end
	end
	do return v:toVectorOfStrings() end
end

function jk.model.DynamicModel:getFieldAsIntegerVector(name)
	local vector = self:getFieldAsDynamicVector(name)
	if not (vector ~= nil) then
		do return nil end
	end
	do
		local v = {}
		do
			local _vl = vector
			if _vl ~= nil then
				local _it = _vl:iterate()
				while _it ~= nil and _it:hasNext() do
					local o = _it:next()
					if o ~= nil then
						goto _continue1
					end
					do _g.jk.lang.Vector:append(v, _g.jk.lang.Integer:asInteger(o)) end
					::_continue1::
				end
			end
		end
		do return v end
	end
end

function jk.model.DynamicModel:getFieldAsLongIntegerVector(name)
	local vector = self:getFieldAsDynamicVector(name)
	if not (vector ~= nil) then
		do return nil end
	end
	do
		local v = {}
		do
			local _vl = vector
			if _vl ~= nil then
				local _it = _vl:iterate()
				while _it ~= nil and _it:hasNext() do
					local o = _it:next()
					if o ~= nil then
						goto _continue2
					end
					do _g.jk.lang.Vector:append(v, _g.jk.lang.LongInteger:asLong(o)) end
					::_continue2::
				end
			end
		end
		do return v end
	end
end

function jk.model.DynamicModel:getFieldAsDoubleVector(name)
	local vector = self:getFieldAsDynamicVector(name)
	if not (vector ~= nil) then
		do return nil end
	end
	do
		local v = {}
		do
			local _vl = vector
			if _vl ~= nil then
				local _it = _vl:iterate()
				while _it ~= nil and _it:hasNext() do
					local o = _it:next()
					if o ~= nil then
						goto _continue3
					end
					do _g.jk.lang.Vector:append(v, _g.jk.lang.Double:asDouble(o)) end
					::_continue3::
				end
			end
		end
		do return v end
	end
end

function jk.model.DynamicModel:getFieldAsBooleanVector(name)
	local vector = self:getFieldAsDynamicVector(name)
	if not (vector ~= nil) then
		do return nil end
	end
	do
		local v = {}
		do
			local _vl = vector
			if _vl ~= nil then
				local _it = _vl:iterate()
				while _it ~= nil and _it:hasNext() do
					local o = _it:next()
					if o ~= nil then
						goto _continue4
					end
					do _g.jk.lang.Vector:append(v, _g.jk.lang.Boolean:asBoolean(o, false)) end
					::_continue4::
				end
			end
		end
		do return v end
	end
end

function jk.model.DynamicModel:addStringToVectorField(field, value)
	do self:getOrCreateFieldAsDynamicVector(field):appendString(value) end
end

function jk.model.DynamicModel:addIntegerToVectorField(field, value)
	do self:getOrCreateFieldAsDynamicVector(field):appendInteger(value) end
end

function jk.model.DynamicModel:addLongIntegerToVectorField(field, value)
	do self:getOrCreateFieldAsDynamicVector(field):appendLong(value) end
end

function jk.model.DynamicModel:addDoubleToVectorField(field, value)
	do self:getOrCreateFieldAsDynamicVector(field):appendDouble(value) end
end

function jk.model.DynamicModel:addBooleanToVectorField(field, value)
	do self:getOrCreateFieldAsDynamicVector(field):appendBoolean(value) end
end

function jk.model.DynamicModel:addBufferToVectorField(field, value)
	do self:getOrCreateFieldAsDynamicVector(field):appendObject(value) end
end

function jk.model.DynamicModel:addObjectToVectorField(field, value)
	do self:getOrCreateFieldAsDynamicVector(field):appendObject(value) end
end

function jk.model.DynamicModel:fromJsonObject(o)
	local data = _vm:to_table_with_key(o, '_isType.jk.lang.DynamicMap')
	if data ~= nil then
		do self:fromDynamicMap(data) end
	else
		do self:clear() end
	end
	do return true end
end

function jk.model.DynamicModel:containsField(field)
	if not (field ~= nil) then
		do return false end
	end
	do
		local array = self:getFields()
		if array ~= nil then
			local n = 0
			local m = _g.jk.lang.Vector:getSize(array)
			do
				n = 0
				while n < m do
					local ff = array[n + 1]
					if ff ~= nil then
						if ff == field then
							do return true end
						end
					end
					do n = n + 1 end
				end
			end
		end
		do return false end
	end
end

function jk.model.DynamicModel:hasValue(key)
	if not (key ~= nil) then
		do return false end
	end
	do return self:getData():containsKey(key) end
end

function jk.model.DynamicModel:removeNullValues()
	if self.dataValue ~= nil then
		do self.dataValue:removeNullValues() end
	end
end

function jk.model.DynamicModel:clear()
	self.dataValue = nil
end

function jk.model.DynamicModel:setAllFields()
	local array = self:getFields()
	if array ~= nil then
		local n = 0
		local m = _g.jk.lang.Vector:getSize(array)
		do
			n = 0
			while n < m do
				local field = array[n + 1]
				if field ~= nil then
					if not self:getData():containsKey(field:getName()) then
						do self:getData():setObject(field:getName(), nil) end
					end
				end
				do n = n + 1 end
			end
		end
	end
end

function jk.model.DynamicModel:getModelMetadata()
	do return nil end
end

function jk.model.DynamicModel:getFieldInformation()
	do return nil end
end

function jk.model.DynamicModel:copyFrom(other, mapping)
	if not (other ~= nil) then
		do return end
	end
	do
		local array = self:getFields()
		if array ~= nil then
			local n = 0
			local m = _g.jk.lang.Vector:getSize(array)
			do
				n = 0
				while n < m do
					local field = array[n + 1]
					if field ~= nil then
						local fname = field:getName()
						local oname = nil
						if mapping ~= nil and mapping:containsKey(fname) then
							oname = mapping:getString(fname, nil)
						else
							oname = fname
						end
						if oname ~= nil and other:hasValue(oname) then
							do self:setField(fname, other:getField(oname)) end
						end
					end
					do n = n + 1 end
				end
			end
		end
	end
end

function jk.model.DynamicModel:getData()
	if not (self.dataValue ~= nil) then
		self.dataValue = _g.jk.lang.DynamicMap._construct0(_g.jk.lang.DynamicMap._create())
	end
	do return self.dataValue end
end

function jk.model.DynamicModel:doSetData(value)
	self.dataValue = value
end

function jk.model.DynamicModel:setData(v)
	do self:doSetData(v) end
	do return v end
end

function jk.model.DynamicModel:getFields()
	if not (self.fieldsValue ~= nil) then
		self.fieldsValue = {}
		do
			local array = self:getFieldInformation()
			if array ~= nil then
				local n = 0
				local m = _g.jk.lang.Vector:getSize(array)
				do
					n = 0
					while n < m do
						local name = array[n + 1]
						if name ~= nil then
							do _g.jk.lang.Vector:append(self.fieldsValue, name) end
						end
						do n = n + 1 end
					end
				end
			end
		end
	end
	do return self.fieldsValue end
end

function jk.model.DynamicModel:doSetFields(value)
	self.fieldsValue = value
end

function jk.model.DynamicModel:setFields(v)
	do self:doSetFields(v) end
	do return v end
end

jk.model.DynamicModelField = {}
jk.model.DynamicModelField.__index = jk.model.DynamicModelField
_vm:set_metatable(jk.model.DynamicModelField, {})

jk.model.DynamicModelField.TYPE_UNKNOWN = 0
jk.model.DynamicModelField.TYPE_OBJECT = 1
jk.model.DynamicModelField.TYPE_STRING = 2
jk.model.DynamicModelField.TYPE_BUFFER = 3
jk.model.DynamicModelField.TYPE_INTEGER = 4
jk.model.DynamicModelField.TYPE_LONG_INTEGER = 5
jk.model.DynamicModelField.TYPE_DOUBLE = 6
jk.model.DynamicModelField.TYPE_BOOLEAN = 7
jk.model.DynamicModelField.TYPE_VECTOR = 8
jk.model.DynamicModelField.TYPE_MAP = 9

function jk.model.DynamicModelField._create()
	local v = _vm:set_metatable({}, jk.model.DynamicModelField)
	return v
end

function jk.model.DynamicModelField:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.model.DynamicModelField'
	self['_isType.jk.model.DynamicModelField'] = true
	self._name = nil
	self._encodedName = nil
	self._type = 0
	self._description = nil
	self._tags = nil
end

function jk.model.DynamicModelField:_construct0()
	jk.model.DynamicModelField._init(self)
	return self
end

function jk.model.DynamicModelField:hasTag(tag)
	local array = self:getTags()
	if array ~= nil then
		local n = 0
		local m = _g.jk.lang.Vector:getSize(array)
		do
			n = 0
			while n < m do
				local tt = array[n + 1]
				if tt ~= nil then
					if tt == tag then
						do return true end
					end
				end
				do n = n + 1 end
			end
		end
	end
	do return false end
end

function jk.model.DynamicModelField:addToTags(tag)
	if not (tag ~= nil) then
		do return end
	end
	if not (self:getTags() ~= nil) then
		do self:setTags({}) end
	end
	do _g.jk.lang.Vector:append(self:getTags(), tag) end
end

function jk.model.DynamicModelField:getName()
	do return self._name end
end

function jk.model.DynamicModelField:doSetName(value)
	self._name = value
end

function jk.model.DynamicModelField:setName(v)
	do self:doSetName(v) end
	do return v end
end

function jk.model.DynamicModelField:getEncodedName()
	do return self._encodedName end
end

function jk.model.DynamicModelField:doSetEncodedName(value)
	self._encodedName = value
end

function jk.model.DynamicModelField:setEncodedName(v)
	do self:doSetEncodedName(v) end
	do return v end
end

function jk.model.DynamicModelField:getType()
	do return self._type end
end

function jk.model.DynamicModelField:doSetType(value)
	self._type = value
end

function jk.model.DynamicModelField:setType(v)
	do self:doSetType(v) end
	do return v end
end

function jk.model.DynamicModelField:getDescription()
	do return self._description end
end

function jk.model.DynamicModelField:doSetDescription(value)
	self._description = value
end

function jk.model.DynamicModelField:setDescription(v)
	do self:doSetDescription(v) end
	do return v end
end

function jk.model.DynamicModelField:getTags()
	do return self._tags end
end

function jk.model.DynamicModelField:doSetTags(value)
	self._tags = value
end

function jk.model.DynamicModelField:setTags(v)
	do self:doSetTags(v) end
	do return v end
end
